src/meltano/core/tracking/iglu-client-embedded/schemas/com.meltano/cli_context/jsonschema/1-0-1

```diff
@@ -0,0 +1,54 @@
+{
+    "$schema": "http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#",
+    "description": "Schema for a Snowplow context describing a Meltano cli command",
+    "self": {
+        "vendor": "com.meltano",
+        "name": "cli_context",
+        "format": "jsonschema",
+        "version": "1-0-1"
+    },
+    "type": "object",
+    "properties": {
+        "command": {
+            "description": "",
+            "type": "string",
+            "maxLength": 1024
+        },
+        "sub_command": {
```

Comment:
Shouldn't `sub_command` be a `$ref` to the schema for `command`, since they can be arbitrarily nested?

Reply:
Unless I'm misunderstanding, they're not arbitrarily nested atm - we're reporting one level.

Reply:
On second thought we probably don't want to report a `sub_command` at all. Instead we probably want multiple command contexts: one for each (nested) command. We could add an `InstrumentedGroup` class that gives us info about a higher-level command, then use `InstrumentedCommand` to give info about the base command.

This would replace the approach that uses `CliContext.from_click_context`. Instead of building a `CliContext` from a Click context, we would have it built inside of `InstrumentedGroup`/`InstrumentedCommand`, and then stored in the Click context as a list under the key `snowplow_contexts` or something like that.

Note that `InstrumentedGroup` would be a subclass of `InstrumentedCommand`, just like how `click.Group` is a suclass of `click.Command`.

Something like `meltano --top-group-level-flag service --mid-group-level-flag start --command-level-flag service_name` would be broken into:
- a command context detailing `meltano --top-group-level-flag`
- a command context for `service --mid-group-level-flag`
- a command context for `start --command-level-flag service_name`

And then with those contexts in place (by `tracker.with_contexts(...)`) the CLI events would fire, and would only contain the event (e.g. `started`, `failed`, `inflight`, etc.). All the other data would be in the contexts.

Reply:
> This also feels like very "late breaking" PR review feedback, I don't know if this approach is objectively better or pedantically better and just pop'd up because this PR's been open so long and you and I've gotten to think about it for so long.

What I described above is what I originally had in mind when I suggested that we hook into Click to get the CLI context. Sorry for the late feedback. This PR fell off my radar, and I've been a bit overwhelmed since - I have not been thinking about this PR until it popped up when you merged `main` into it.

I understand your reservations with the proposed change, and agree that at some point we need to accept what we have got and move on. That said, with the current approach, how can we deal with nested commands that have more than a single layer? I suppose we can use `CliContext.from_click_context::_recursively_collect_params` to collect it all, then flatten it, but I'm concerned that flattening it will make use lose valuable information. We can't associate options/arguments/flags with the command/sub-command that they were applied to.

The current schema has a `sub_command` field, but that would only work up to one layer deep if not for the special case in place (`if ctx.parent and ctx.parent.command.name != "cli":`) which lets it work two layers deep.

Since the commands have an inherently nested structure imposed by Click's concepts of groups/commands, the data we collect from them should also be nested. Whether it goes into one `CliContext` with a nested schema, or multiple with flat schemas does not matter to me, but the loss of information and inconsistency does.